src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadata.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata;
+
+/**
+ * Responsible for storing metadata of an argument.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+class ArgumentMetadata
+{
+    private $name;
+    private $type;
+    private $isVariadic;
+    private $hasDefaultValue;
+    private $defaultValue;
+
+    public function __construct($name, $type, $isVariadic, $hasDefaultValue, $defaultValue)
+    {
+        $this->name = $name;
+        $this->type = $type;
+        $this->isVariadic = $isVariadic;
+        $this->hasDefaultValue = $hasDefaultValue;
+        $this->defaultValue = $defaultValue;
+    }
+
+    /**
+     * Returns the name as given in PHP, $foo would yield "foo".
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * The PHP class in 5.5+ and additionally the basic type in PHP 7.0+.
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * If the argument is defined as "...$variadic".
+     *
+     * @return bool
+     */
+    public function isVariadic()
+    {
+        return $this->isVariadic;
+    }
+
+    /**
+     * If the argument has a default value.
+     *
+     * Implies an optional argument when true.
```

Comment:
`Returns whether the argument has a default value.`

`Returns whether...` seems to be a kind of convention in core for returning a bool.